lib/connect.ts

```diff
@@ -65,1354 +69,648 @@ import utils from "./utils";
+        return this._get('portfolio.holdings');
     };
 
-    /**
-     * Retrieves list of available instruments for a auction session.
-     * @method getAuctionInstruments
-     * @memberOf KiteConnect
-     * @instance
-     */
-    self.getAuctionInstruments = function() {
-        return _get("portfolio.holdings.auction");
-    }
+    getAuctionInstruments(): Promise<AxiosResponse<any>> {
+        return this._get('portfolio.holdings.auction');
+    };
 
-    /**
-     * Retrieve positions.
-     * @method getPositions
-     * @memberOf KiteConnect
-     * @instance
-     */
-    self.getPositions = function() {
-        return _get("portfolio.positions");
+    getPositions(): Promise<AxiosResponse<any>> {
+        return this._get('portfolio.positions');
     };
 
-    /**
-     * Modify an open position's product type.
-     * @method convertPosition
-     * @memberOf KiteConnect
-     * @instance
-     * @param {Object} params params.
-     * @param {string} params.exchange Exchange in which instrument is listed (NSE, BSE, NFO, BFO, CDS, MCX).
-     * @param {string} params.tradingsymbol Tradingsymbol of the instrument  (ex. RELIANCE, INFY).
-     * @param {string} params.transaction_type Transaction type (BUY or SELL).
-     * @param {string} params.position_type Position type (overnight, day).
-     * @param {string} params.quantity Position quantity
-     * @param {string} params.old_product Current product code (NRML, MIS, CNC).
-     * @param {string} params.new_product New Product code (NRML, MIS, CNC).
-     */
-    self.convertPosition = function(params) {
-        return _put("portfolio.positions.convert", params);
+    convertPosition(params: Order): Promise<AxiosResponse<any>> {
+        return this._put('portfolio.positions.convert', params);
     };
 
-    /**
-     * Retrieve the list of market instruments available to trade.
-     * Note that the results could be large, several hundred KBs in size,
-     * with tens of thousands of entries in the list.
-     * Response is array for objects. For example
-     * ~~~~
-     * 	{
-     * 		instrument_token: '131098372',
-     *		exchange_token: '512103',
-     *		tradingsymbol: 'NIDHGRN',
-     *		name: 'NIDHI GRANITES',
-     *		last_price: '0.0',
-     *		expiry: '',
-     *		strike: '0.0',
-     *		tick_size: '0.05',
-     *		lot_size: '1',
-     *		instrument_type: 'EQ',
-     *		segment: 'BSE',
-     *		exchange: 'BSE' }, ...]
-     * ~~~~
-     *
-     * @method getInstruments
-     * @memberOf KiteConnect
-     * @instance
-     * @param {Array} [segment] Filter instruments based on exchange (NSE, BSE, NFO, BFO, CDS, MCX).
-     * If no `segment` is specified, all instruments are returned.
-     */
-    self.getInstruments = function(exchange) {
-        if(exchange) {
-            return _get("market.instruments", {
-                "exchange": exchange
+    getInstruments(exchange: boolean): Promise<AxiosResponse<any>> {
```

Comment:
The incorrect exchange type is being used; it should be a `string` instead of a `boolean`.